### PR TITLE
update workflow files

### LINKED ISSUE #1203: Feature: Improve CI
## Feature Details
- Evaluate CI and decide what is useful and what is burdensome
- Look for ways to cut back on runtime without affecting code quality/early flagging of code issues

### LINKED ISSUE #1280: Improvement to reduce dependencies
## QC Details
Use `isNamespaceLoaded()` like [here](https://github.com/SafetyGraphics/safetyGraphics/blob/f602cdc2068e7b290513c9accad7cf351775ce4e/R/mod_profileTab.R#L10) in reporting to only render certain elements (e.g., the study timeline) if packages are available and loaded.

This will allow us to move `ggiraph`, `gggenes`, and maybe a few other dependencies to `Suggests` in the `DESCRIPTION` file.

```
pkgndep::dependency_heatmap(pkgndep::pkgndep('gsm'), file = "gsm_dependencies.png")
```

Image

### LINKED ISSUE #1468: QC: Update year on license
## QC Details
[It's 2024](https://github.com/Gilead-BioStats/gsm/blob/dev/LICENSE.md?plain=1#L182)!

### PR DESCRIPTION
**Pull request by mattroumaya**
## Overview

A lot of this housekeeping PR is dealing with moving dependencies to `Suggests:`, following the guidance in [this chapter](https://r-pkgs.org/dependencies-in-practice.html#sec-dependencies-in-imports-r-code).
- From now on, we should not use `@importFrom package function` in roxygen, but instead use `usethis::use_import_from("package", "function")`, which will add the equivalent `@importFrom()` statement to [`gsm-package.R`](https://github.com/Gilead-BioStats/gsm/blob/7c62285bf27ddda0a6e476852e9b252a6f68e4c4/R/gsm-package.R)
- A fair amount of reporting packages were moved to `Suggests:`, and `rlang::check_installed()` will see if the package is available, and if not, it will prompt the user to install the necessary packages.
  - On a system-level deployment/installation, as long as `{gsm}` is installed with `dependencies = TRUE`, there should be no issues.

Fix #1203
Fix #1468
Fix #1280

### CODE REVIEW

File: R/Disp_Map_Raw.R [hunk 1]
```diff
@@ -34,19 +34,14 @@
...
-#' @import dplyr
-#'
 #' @export
 
 Disp_Map_Raw <- function(
   dfs = gsm::UseClindata(
     list(
       "dfSUBJ" = "clindata::rawplus_dm",
       "dfSTUDCOMP" = "clindata::rawplus_studcomp",
-      "dfSDRGCOMP" = "clindata::rawplus_sdrgcomp %>% filter(.data$phase == 'Blinded Study Drug Completion')"
```
Comment (samussiah) [inline R/Disp_Map_Raw.R:49]: Not clear why this filter is being removed.

File: R/Make_Snapshot.R [hunk 2]
```diff
@@ -55,8 +49,7 @@ Make_Snapshot <- function(
       "dfIE" = "clindata::rawplus_ie",
       "dfLB" = "clindata::rawplus_lb",
       "dfSTUDCOMP" = "clindata::rawplus_studcomp",
-      "dfSDRGCOMP" = "clindata::rawplus_sdrgcomp %>%
```
Comment (samussiah) [inline R/Make_Snapshot.R:58]: Not clear why this filter is being removed.

File: R/Study_Assess.R [hunk 3]
```diff
@@ -55,7 +49,8 @@ Study_Assess <- function(
         "dfIE" = "clindata::rawplus_ie",
         "dfLB" = "clindata::rawplus_lb",
         "dfSTUDCOMP" = "clindata::rawplus_studcomp",
-        "dfSDRGCOMP" = "clindata::rawplus_sdrgcomp %>% dplyr::filter(.data$phase == 'Blinded Study Drug Completion')",
```
Comment (samussiah) [inline R/Study_Assess.R:58]: But not here?

File: R/gsm-package.R
Comment (samussiah) [inline R/gsm-package.R:1]: centralized namespacing  baby

**Review by samussiah — APPROVED** [review]
Hella organized @mattroumaya - love it, and thanks for doing the research on best practices.  It's interesting they don't mention including the function import in the header block of the file (the approach I've seen most commonly) - only above the function call or in a central location.

**Comment by samussiah** [comment]
One other question - will `check()` determine whether a function is not namespaced?  Well, I guess all external functions  will be `::`'d going forward.

**Comment by mattroumaya** [comment]
> One other question - will `check()` determine whether a function is not namespaced? Well, I guess all external functions will be `::`'d going forward.

If we add a new function from `{pkgFoo}` and don't add it to the `NAMESPACE`, R CMD Check should throw an error like this:

```
❯ checking dependencies in R code ... WARNING
  '::' or ':::' import not declared from: pkgFoo
```

**Pull request merged**